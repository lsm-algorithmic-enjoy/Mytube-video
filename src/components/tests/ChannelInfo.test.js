import { render, screen } from "@testing-library/react";
import { Route } from "react-router-dom";
import { withAllContexts, withRouter } from "../../tests/utils";
import ChannelInfo from "../ChannelInfo";

describe("ChannelInfo", () => {
  const fakeYoutube = { channelImageURL: jest.fn() };

  afterEach(() => fakeYoutube.channelImageURL.mockReset());

  it("renders correctly", async () => {
    fakeYoutube.channelImageURL.mockImplementation(() => "url");
    const { asFragment } = renderChannelInfo();
    await screen.findByRole("img");
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without URL", () => {
    fakeYoutube.channelImageURL.mockImplementation(() => {
      throw new Error("error");
    });
    renderChannelInfo();
    expect(screen.queryByRole("img")).toBeNull();
  });

  it("renders with URL", async () => {
    fakeYoutube.channelImageURL.mockImplementation(() => "url");
    renderChannelInfo();
    await screen.findByRole("img");
  });
  function renderChannelInfo() {
    return render(
      withAllContexts(
        withRouter(
          <Route path="/" element={<ChannelInfo id="id" name="channel" />} />
        ),
        fakeYoutube
      )
    );
  }
});
