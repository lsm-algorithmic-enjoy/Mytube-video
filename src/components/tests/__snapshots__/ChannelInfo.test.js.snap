// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex my-4 mb-8 items-center"
  >
    <img
      alt="channel"
      class="w-10 h-10 rounded-full"
      src="url"
    />
    <p
      class="text-lg font-medium ml-2"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;
